### DISCUSSION TITLE
How to manage multiple requestBody in a single endpoint

### DISCUSSION THREAD

**Discussion by ydevmob** [OAI/OpenAPI-Specification#2974] [Q&A]
Is it possible to have more than one requestBody for the same endpoint post;
`
        "Entete": {
            "operationid": 7701,
        },
"contentParams":{
"user":"username",
"password":"passworduser"
}
`
This a piece of my request body and depending of the operationid  execute a method , 
Note: operation code are unique for every method;
More info: contentParams depend of the operationid,
every operationid have its unique id, but the content params might be same with other methods but the output is always different;
If i am not clear enough i can explain more, and ask me question when need be;

**Comment by Wyze5280** [comment]
No Idea what that means please email me for more details ceelo19888@gmail.com

**Comment by rafalkrupinski** [comment]
operationId is just a name for a unique pair of path and HTTP method.

Each unique triplet of method, path and content-type declares a separate schema for request body. Furthermore that schema can be defined as `oneOf` and accept a list of alternative schemas, or it could be an object schema with properties that are also schemas. There are other options, but these two seem to be related to your question.

HTH

**Comment by handrews** [comment]
It is not possible in OAS 3, but it is being added to OAS 4 "Moonwalk" (see discussions in the OAI/sig-moonwalk repository).  Closing in favor fo the Moonwalk discussions.
